### PR TITLE
feat:  Getting started with React (#40385)

### PR DESCRIPTION
**Pull request by vishalmalhotra1998**
[Issue. No: 40385](https://redmine.successive.tech/issues/40385)

Subject-
1. Add TextFieldDemo.jsx
2. Add TextField.jsx

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
+charset = utf-8
+trim_trailing_whitespace = false
+insert_final_newline = false
```
Comment (madhavBansal-successiveTech) [inline .editorconfig]: fix lint error

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      "import/prefer-default-export": "off"
+    },
+    "parserOptions": {
+      "ecmaVersion": 6,
+      "ecmaFeatures": {
+        "jsx": true,
+        "classes": true
+      }
+    }
+  }
```
Comment (madhavBansal-successiveTech) [inline .eslintrc]: fix lint error

File: .huskyrc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "hooks": {
+    "pre-commit": "npm run lint && npm run test:c",
+    "pre-push": "npm run lint && npm run test:c"
+  }
+}
```
Comment (madhavBansal-successiveTech) [inline .huskyrc]: fix lint error

File: src/App.jsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import TextEditor from './pages/index';
```
Comment (madhavBansal-successiveTech) [inline src/App.jsx]: did not use index for import - use (import TextEditor from './pages')
export all pages from index.js

File: src/components/TextField/TextField.jsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { Input, Error } from './style';
+
+const TextField = (values = '') => {
+  const { value, disabled, error } = values;
+  return (
+    <>
+      <Input type="text" value={value} disabled={disabled} />
+      {error ? <Error>{error}</Error> : <br />}
```
Comment (madhavBansal-successiveTech) [inline src/components/TextField/TextField.jsx]: instead use of br in ternaray operator use null or blank string like - 
error ? <Error>{error}</Error> : ''

File: src/components/TextField/index.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+import TextFieldDemo from './TextField';
+
+export default TextFieldDemo;
```
Comment (madhavBansal-successiveTech) [inline src/components/TextField/index.js]: export in single line

File: src/components/index.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import TextFieldDemo from './TextField/index';
+
+export default TextFieldDemo;
```
Comment (madhavBansal-successiveTech) [inline src/components/index.js]: export in single line

File: src/pages/TextFieldDemo/TextFileDemo.jsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import TextField from '../../components/index';
+import { Paragraph } from '../../components/TextField/style';
+
+const TextEditor = () => (
+  <>
+    <Paragraph>This is a disable input</Paragraph>
+    <TextField value="Disabled Input" disabled="true" />
```
Comment (madhavBansal-successiveTech) [inline src/pages/TextFieldDemo/TextFileDemo.jsx]: for pass boolean true value we didn't use the value in string just pass disabled as props like (disabled)

File: src/pages/TextFieldDemo/index.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+import TextEditor from './TextFileDemo';
+
+export default TextEditor;
```
Comment (madhavBansal-successiveTech) [inline src/pages/TextFieldDemo/index.js]: use single export

**Review by madhavBansal-successiveTech — COMMENTED** [review]
Fix linting error and removes some unwanted files like serviceWorker.js, App.test.js etc

File: .eslintrc (hunk 2 above)
Comment (vishalmalhotra1998) [reply .eslintrc]: Done

File: src/App.jsx (hunk 4 above)
Comment (vishalmalhotra1998) [reply src/App.jsx]: Done

File: src/components/TextField/TextField.jsx (hunk 5 above)
Comment (vishalmalhotra1998) [reply src/components/TextField/TextField.jsx]: Done

File: src/pages/TextFieldDemo/TextFileDemo.jsx (hunk 8 above)
Comment (vishalmalhotra1998) [reply src/pages/TextFieldDemo/TextFileDemo.jsx]: Done

File: src/components/TextField/index.js (hunk 6 above)
Comment (vishalmalhotra1998) [reply src/components/TextField/index.js]: Done

File: .editorconfig (hunk 1 above)
Comment (vishalmalhotra1998) [reply .editorconfig]: Done

**Pull request merged**